hack/test-extended.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+source ${OS_ROOT}/hack/util.sh
+source ${OS_ROOT}/hack/common.sh
+
+# TODO: Randomize these ports
```

Comment:
Is the reason for using random ports that you could run the tests in parallel on a single machine?